Utilities/rANS/include/rANS/internal/SymbolTable.h

```diff
@@ -77,24 +95,30 @@ template <typename T>
 inline const T& SymbolTable<T>::operator[](int64_t index) const
 {
   const auto idx = index - mMin;
-  assert(idx >= 0);
-  assert(idx < mSymbols.size());
-  return *(mIndex[idx]);
+  if (idx >= 0 || idx < mIndex.size()) {
```

Comment:
here there should be ``if (idx >= 0 && idx < mIndex.size())``
But I think one can have just 1 if by doing
````
uint64_t idx = static_cast<uin64_t>(index - mMin);
return (idx<mIndex.size()) ? *(mIndex[idx]) : *mEscapeSymbol;
````

Reply:
you forgot ``return``. But why not using just one if comparison instead of 2?

Reply:
> @MichaelLettrich thanks, please rebase to dev and see comment below.
> @davidrohr this PR will supersede the #4123 but will require dictionaries regenerated, I'll close mine once this one is ready to merge.

@davidrohr @shahor02 : Dictionaries stay valid. The only difference is how the symbol tables are generated from it.

Reply:
Because the symbol table stores symbols in the range `[mMin,mMax]`, so we have to treat both cases for a rare symbol being `> mMax` or `<mMin`, which requires two comparisons.

Reply:
you compare not ``mMin<=symbol<=mMax`` but ``idx=index - mMin`` being <0 or >size. If you static_cast negative idx to unsigned, it will be > MAX_INT, so the comparison ``static_cast<uin64_t>(index - mMin)<mIndex.size()) ...`` will cover both cases.